### DISCUSSION TITLE
Prisma MySQL + Hostinger problems.

### DISCUSSION THREAD

**Discussion by pDallastra** [prisma/prisma#13600] [Q&A]
I was using Prisma with Postgres and it was amazing! I could connect it easily with my free Heroku server and all of that.

I had to buy a host service that only allows me to use MySQL, but I have not been able to connect it to it. I am not sure if it is a problem related to my hoster, or if it is related to how Prisma is trying to connect to it. I was able to connect to my hoster by using DBeaver and TypeORM without any problem.

Something I notice is that I set a password to a database in the server, but the password that allows getting a connection is the one I created for my "login" to the hoster, not the one I created for the database. 

The steps I tried and the errors I got:

1 - Right database password - Error with Authentication
2 - Using my hoster user password as the database password - unable to reach the database.

Again, I am able to connect to the database using TypeORM and DBeaver.

**Comment by austincrim** [comment]
Hey @pDallastra 👋,

Could you share the full error messages and codes? Does your database user have full access?

**Reply by pDallastra** [reply]
@austincrim thanks for taking the time to look at this man.

DATABASE_URL=mysql://u522094498_dallastra:<mypassword>@sql803.main-hosting.eu:3306/u522094498_impext

Something I just noticed is data source DB is saying MySQL database is "undefined"... But I think I have put everything placed as shown in the docs.

While building:
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": MySQL database "undefined" at "u522094498_dallastra:3306"

After the attempt:
Error: P1001: Can't reach database server at `u522094498_dallastra`:`3306`
Please make sure your database server is running at `u522094498_dallastra`:`3306`.

It might be related to that undefined part, but I don't see why it is getting undefined for the DATABASE.

**Reply by austincrim** [reply]
Hmm, that logging definitely seems off. It's displaying the username where the host should go and the db name is coming back `undefined`. It's messing up parsing the connection string somehow.

Could you fill out a [bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) with more details about your environment and setup? Our engineering teams can look at it from there. Thanks!

**Reply by pDallastra** [reply]
Done.
I will wait until they get it fixed, and then I will consider this one fixed.

The Bug Report link:
https://github.com/prisma/prisma/issues/13622

**Comment by stibenvelez** [comment]


**Reply by jonahallibone** [reply]
@stibenvelez at the risk of alerting other people, your comment seems to contain sensitive information about your database. You might want to edit or remove your comment as to not expose those further, and also change the credentials on your database to minimize the chances of any abusive actions.

**Comment by stibenvelez** [comment]
@austincrim heeelp!!!
I have been trying to connect my database with Prisma for several days. It has not been possible, I have been trying for days. The credentials are correct, I can login even from TablePlus or WorkBench. I hope you can help me!

Thank you

PrismaClientInitializationError:
Invalid prisma.productCategory.count() invocation:

Authentication failed against database server at sql452.main-hosting.eu, the provided database credentials for u189463349_user_food are not valid.

Please make sure to provide valid database credentials for the database server at sql452.main-hosting.eu.

datasource db {
provider = "mysql"
url = "mysql://u189463349_user:X%[23W@sql452.main-hosting.eu](mailto:23W@sql.main-hosting.eu):3606/u1894app"
}

**Reply by austincrim** [reply]
Are you doing any URL percent encoding for the database password? Special characters need to be encoded, including the percent character itself.

You can check out [this table](https://developer.mozilla.org/en-US/docs/Glossary/percent-encoding) as a reference.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
